### PR TITLE
feat: add implementation for TopsortClient

### PR DESCRIPTION
**Pull request by barbmarcio**
This PR implements the TopsortClient as part of the configuration initialization.
It also includes an endpoint sanitizer to prevent trailing slashes from passing through.

Closes: https://topsort.atlassian.net/browse/API-1147

### CODE REVIEW

File: src/functions/events.ts [hunk 1]
```diff
@@ -5,33 +5,11 @@ import { withValidation } from "../lib/with-validation";
...
-  let url: URL;
+async function handler(event: TopsortEvent, config: Config): Promise<EventResult> {
   try {
-    url = new URL(apis.events, config.host || baseURL);
-  } catch (error) {
-    throw new AppError(400, "Invalid URL", {
-      error: `Invalid URL: ${error}`,
-    });
-  }
+    const url = `${config.host}${apis.events}`;
```
Comment (jbergstroem) [inline src/functions/events.ts:10]: wouldn't baseUrl come through config?

File: src/functions/topsort-client.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  constructor(config: Config) {
+    this.config = config;
+    this.config.host = this.sanitizeUrl(this.config.host ?? baseURL);
+  }
+
+  private sanitizeUrl(url: string): string {
+    try {
+      const parsedUrl = new URL(url);
+      return parsedUrl.href.replace(/\/+$/, "");
```
Comment (jbergstroem) [inline src/functions/topsort-client.ts:25]: I don't think it makes sense to do this until we merge it with path in the future

Comment (barbmarcio) [reply src/functions/topsort-client.ts:25]: Yep, I just changed the function, will push in a bit

File: src/functions/events.ts (hunk 1 above)
Comment (barbmarcio) [reply src/functions/events.ts]: Yes, I was thinking only the baseURL of the API, then for each function we would put the path in

**Comment by jbergstroem** [comment]
- Please update README!
- Add a test for auction or event that verifies that the url in the call made is stripped of a trailing slash

File: README.md [hunk 3]
```diff
@@ -31,10 +31,10 @@ yarn add @topsort/sdk --save
 
 ### Auctions
 
-To create an auction, use the `createAuction` function. Example:
+To create an auction, first initialize Topsort Client, then use the `createAuction` function. Example:
```
Comment (jbergstroem) [inline README.md:34]:
```suggestion
To create an auction, first initialize a Topsort Client, then call the `createAuction` function:
```

File: README.md [hunk 4]
```diff
@@ -31,10 +31,10 @@ yarn add @topsort/sdk --save
...
 
 ```js
-import { TopsortAuction, createAuction } from "@topsort/sdk";
+import { TopsortClient, TopsortAuction } from "@topsort/sdk";
```
Comment (jbergstroem) [inline README.md:37]: Can't help but feel there's a lot of "Topsort" everywhere. Do we really need it? We're already in `@Topsort/sdk`.

File: README.md [hunk 5]
```diff
@@ -59,11 +59,13 @@ const config = {
   // generate your api key in the auction manager - it should look some thing like this
   // note: this is an invalid key and won't work, you need to replace it with your own
   apiKey: "TSE_4S6o1g1CB5tyRENfhDMAn6viR7A5cy3j1JAR",
-  userAgent: "Mozilla/5.0"
-  timeout: 50
+  userAgent: "Mozilla/5.0", // optional user agent to be added as part of the request
+  timeout: 50 // optional timeout signal to be added asp art of the request
```
Comment (jbergstroem) [inline README.md:63]:
```suggestion
  timeout: 50 // optional timeout for the request
```

File: README.md [hunk 6]
```diff
@@ -118,10 +120,10 @@ createAuction(config, auctionDetails)
 
 ### Events
 
-To report an event, use the reportEvent function. Here is an example:
+To report an event, first initialize Topsort Client, then use the `reportEvent` function. Here is an example:
```
Comment (jbergstroem) [inline README.md:123]:
```suggestion
To report an event, first initialize a Topsort Client, then call the `reportEvent` function:
```

File: src/functions/events.ts [hunk 7]
```diff
@@ -1,37 +1,15 @@
-import { apis, baseURL } from "../constants/apis.constant";
+import { apis } from "../constants/apis.constant";
```
Comment (jbergstroem) [inline src/functions/events.ts:1]: I think `endpoints` is a better name than `apis`

**Review by jbergstroem — CHANGES_REQUESTED** [review]
Since we're breaking the api we might as well do some name changes. Thoughts?

**Pull request merged**